lib/sidekiq_unique_jobs/digests.rb

```diff
@@ -41,17 +41,27 @@ def delete_by_pattern(pattern, count: DEFAULT_COUNT)
     #   Also deletes the :AVAILABLE, :EXPIRED etc keys
     #
     # @param [String] digest a unique digest to delete
-    def delete_by_digest(digest) # rubocop:disable Metrics/MethodLength
+    # @param queuetime [Boolean] Whether to delete queue locks.
+    # @param runtime [Boolean] Whether to delete run locks.
+    def delete_by_digest(digest, queuetime: true, runtime: true) # rubocop:disable Metrics/MethodLength
       result, elapsed = timed do
+        queue = queuetime ? digest : ""
+        queue_queued = queuetime ? "#{digest}:QUEUED" : ""
```

Comment:
I'm still thinking about this, I get the problem, but wondering if there is a better way. 

The code ended up being a little repetitive below. Perhaps we could just have two other methods:

1. queuetime_keys
2. runtime_keys

```ruby
keys = []
keys.concat(queuetime_keys)
keys.concat(runtime_keys)

call_script(:delete_by_digest, keys)

EMPTY_KEYS_SEGMENT = ["", "", "", ""].freeze

def queuetime_keys(queuetime)
  return EMPTY_KEYS_SEGMENT unless queuetime
  
  [
    digest,
    "#{digest}:QUEUED",
    "#{digest}:PRIMED",
    "#{digest}:LOCKED",  
  ]
end
```

What do you think? I find the code a little harsh on the eyes. Do you know any better way (than mine) to construct the array to send to the script (the keys) I'm all ears and eyes.

Reply:
I refactored the code a little and created the two helper methods. WDYT?